Backend/Controllers/DTOs/Groups.cs

```diff
@@ -0,0 +1,43 @@
+using System.ComponentModel.DataAnnotations;
+
+namespace Backend.Controllers.DTOs;
+
+public class PostGroupDTO
+{
+    /// <inheritdoc cref="Models.Group.Name"/>
+    [StringLength(100)]
+    public required string Name { get; set; }
+
+    /// <inheritdoc cref="Models.Group.Type"/>
+    public required Models.GroupType Type { get; set; }
+}
+
+public class GroupResponseDTO
+{
+    /// <inheritdoc cref="Models.Group.Id"/>
+    public uint Id { get; set; }
+
+    /// <inheritdoc cref="Models.Group.Name"/>
+    public required string Name { get; set; }
+
+    /// <inheritdoc cref="Models.Group.Active"/>
+    public bool Active { get; set; } = true;
+
+    /// <inheritdoc cref="Models.Group.GroupMemberships"/>
+    public List<GroupMembershipResponseDTO> GroupMemberships { get; set; } = new();
```

Comment:
Perhaps change this to a list of members who are in this group instead of whole memberships? Depends on what the frontend wants though, so this is subjective. This change will simplify the function where these DTOs are made though